dartagnan/src/main/java/com/dat3m/dartagnan/program/event/lang/svcomp/NonDetChoice.java

```diff
@@ -0,0 +1,46 @@
+package com.dat3m.dartagnan.program.event.lang.svcomp;
+
+import com.dat3m.dartagnan.program.Register;
+import com.dat3m.dartagnan.program.event.AbstractEvent;
+import com.dat3m.dartagnan.program.event.Event;
+import com.dat3m.dartagnan.program.event.RegWriter;
+import com.google.common.base.Preconditions;
+
+public class NonDetChoice extends AbstractEvent implements RegWriter {
+
+    protected Register register;
+    protected boolean isSigned;
+
+    public NonDetChoice(Register register, boolean isSigned) {
+        this.register = Preconditions.checkNotNull(register);
+        this.isSigned = isSigned;
+    }
+
+    protected NonDetChoice(NonDetChoice other) {
+        super(other);
+        this.register = other.register;
+        this.isSigned = other.isSigned;
+    }
+
+    public boolean isSigned() { return isSigned; }
+
+    @Override
+    protected String defaultString() {
+        return String.format("%s <- *", register);
```

Comment:
Why not using a more self explaining string, like `non-det()`?

Reply:
I didn't wanna use anything with parenthesis `()` cause it looks like a function call (and this is not call). Without the parenthesis, `r <- non_det` looks the same as a standard register assignment, so it was not an option either.
So `*` ended up being the most clear choice while also being standard notation.